### PR TITLE
Various fixes for bulk-upload code

### PR DESCRIPTION
**Pull request by pbanaszkiewicz**
This should fix most of the bugs from https://github.com/swcarpentry/amy/pull/201

Please check and merge so that we can continue discussion under https://github.com/swcarpentry/amy/pull/201.

Commit messages contain most of the explanation.

### CODE REVIEW

File: workshops/test/test_util.py [hunk 1]
```diff
@@ -131,7 +131,7 @@ def test_verify_event_dne(self):
         self.assertTrue(len(errors) == 1)
         self.assertTrue('Event with slug' in errors[0])
 
-    def test_very_role_dne(self):
+    def test_verify_role_dne(self):
```
Comment (gvwilson) [inline workshops/test/test_util.py]: What is 'dne' ?

Comment (pbanaszkiewicz) [inline workshops/test/test_util.py]: It stands for "Does Not Exist".

Comment (gvwilson) [inline workshops/test/test_util.py]: Ah.

On 2015-02-25 2:57 PM, Piotr Banaszkiewicz wrote:
>
> In workshops/test/test_util.py 
> <https://github.com/gvwilson/amy/pull/1#discussion_r25373097>:
>
> > @@ -131,7 +131,7 @@ def test_verify_event_dne(self):
> >          self.assertTrue(len(errors) == 1)
> >          self.assertTrue('Event with slug' in errors[0])
> >
> > -    def test_very_role_dne(self):
> > +    def test_verify_role_dne(self):
>
> It stands for "Does Not Exist".
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/gvwilson/amy/pull/1/files#r25373097>.
>

-- 
Email longa, vita brevis.

Comment (wking) [inline workshops/test/test_util.py]: In my experience, abbreviating function names is rarely worth it.  You save a few keystrokes here, but it's not like you're typing this function name all over the place.  And having the full wording in the function name will give you more obvious test output.  If @gvwilson wasn't clear on the meaning while he was looking at the code, it's likely that a future tester will be confused if that test fails to pass and all she sees are the abbreviated function name and a traceback.

**Pull request merged**